xdem/terrain.py

```diff
@@ -776,9 +1253,9 @@ def get_terrain_attribute(
     :param hillshade_z_factor: Vertical exaggeration factor.
     :param slope_method: Method to calculate the slope, aspect and hillshade: "Horn" or "ZevenbergThorne".
     :param tri_method: Method to calculate the Terrain Ruggedness Index: "Riley" (topography) or "Wilson" (bathymetry).
-    :param fill_method: See the 'get_quadric_coefficients()' docstring for information.
-    :param edge_method: See the 'get_quadric_coefficients()' docstring for information.
     :param window_size: Window size for windowed ruggedness and roughness indexes.
+    :param engine: Engine to use for computing the attributes with convolution or other windowed calculations, currently
```

Comment:
Maybe the choice of engine should be slightly more documented, here or in the doc? Basically, what could be reasons for chosing scipy over the default numba?
From your benchmark tests, I see that scipy is faster for small arrays, but does it matter if the array is small anyway and computation time is small? Are there other reasons that could push a user to use scipy?

Reply:
I've added an explanation in the doc.
What you see for small arrays is actually the initial compile time of Numba, which is unavoidable. I've explained that.

Reply:
I basically didn't do a first independent call to cache before timing the performance, so the first array size is wrong in the plot.
See: https://numba.readthedocs.io/en/stable/user/5minguide.html#how-to-measure-the-performance-of-numba